crates/bevy_ecs/src/schedule/executor/multi_threaded.rs

```diff
@@ -483,12 +555,14 @@ impl MultiThreadedExecutor {
             }
 
             // Evaluate the system set's conditions.
+            // SAFETY: We have exclusive access to ExecutorState, so no other thread is touching these conditions
```

Comment:
This is not guaranteed given the function signature and the safety invariants, this likely needs to be propagated through the invariants of this function.